### ISSUE TITLE
[Button] - Disabled icon only button has incorrect pointer

### ISSUE THREAD

**Issue by tomkdgun** [carbon-design-system/carbon#4475]
[Button] - Disabled icon only button has incorrect pointer

Standard disabled button has cursor defined as `cursor:not-allowed;`.
At the same time disabled icon only button (`hasIconOnly: true`) has 
`cursor:pointer;`

## Step to reproduce:
1. Go to http://react.carbondesignsystem.com/?path=/story/buttons--icon-only-buttons
2. Select disabled prop on Knobs.

## What package(s) are you using?

    "carbon-components": "10.7.0",
    "carbon-components-react": "7.7.0",
    "carbon-icons": "7.0.7",

**Comment by jendowns** [comment]
@dakahn I'm curious what you think about this --

Aside from the cursor issue, disabled buttons get the `disabled` attribute & can't be focused. So the tooltip being added to the icon-only button, in this case, can't be accessed by a keyboard user if the icon-only button is also `disabled`. Yet on hover, someone with a mouse can still see the tooltip (and that pointer cursor, as stated in the issue desc).

So what should the consistent experience be here? It seems important to show a visible tooltip, whether `disabled` or not, the same way you'd be able to see the button text of an ordinary, non-icon-only button? 🤔 But with `disabled`, the button cannot be focused...

**Comment by tomkdgun** [comment]
@jendowns @dakahn In general icon only buttons tooltips are problematic. Currently last clicked button has focus, so tooltip is visible. When I hover over another icon button, another tooltip is also visible, so both are present and overlapping each other:

![image](https://user-images.githubusercontent.com/17591704/68208412-5fae0a80-ffd1-11e9-84e6-476df6e9af95.png)


Steps to recreate:
1. Create two icon only buttons next to each other.
2. Click on button one - button focused tooltip visible,
   or focus button using Tab
3. Hover over button two - both tooltips visible and in most cases overlaps since buttons are 48px size.

**Comment by dakahn** [comment]
@jendowns That's a real head scratcher. Ideally (and you probably know this) icon-only buttons should be avoided unless _absolutely necessary_ (like when space is at an absolute premium), for the reasons like the one @tomkdgun talked about and a whole host of others as well. 😞 

My gut says that if we're going to rely on a tooltip to describe a button that disabled state is something that should be accessible to everyone. So we maybe don't use the `disabled` attribute and instead disable the buttons functionality and have it describe itself like "button, submit, disabled" so it can still be focusable and communicate to keyboard users etc. 

Staring at the icons at the top of my Firefox browser I can see that from this page I can go back. So I can use my keyboard to navigate to the back button and press it, but I cannot go forward (because there is no forward history) -- this button is disabled and is not accessible with my keyboard. Though I can hover over it with my pointer and get a tooltip. 

@snidersd What has been IBMa's guidance on this issue in the past?

**Comment by tomkdgun** [comment]
@dakahn Icon only buttons are very common to be used on toolbars, or as buttons inside DataTable rows e.g. edit/delete actions.
Regarding accessibility, use of aria-label should be sufficient. I think screen readers will also correctly handle disabled button state, without a need of adding aria-disabled.

Set of icon only buttons are common in toolbars (just different icons):
![image](https://user-images.githubusercontent.com/17591704/68208570-b9aed000-ffd1-11e9-91d0-3001e5df335b.png)

**Comment by tomkdgun** [comment]
Any updates on this issue ?

**Comment by dakahn** [comment]
@tomkdgun your aria solution covers keyboard users with a screen reader -- but not keyboard only users who may _need to know_ a particular control is disabled (and possibly why).

**Comment by dakahn** [comment]
I'm going to close this issue out since we're not pursuing any solutions here and it seems like the issue has gone stale. 

In our design system if a form control is disabled due to some user action/inaction they should be made well aware of that through an announcement or notification of some kind. Relying on the button to communicate its disabled state is problematic, but there are maybe ways of getting this done if we want to open an enhancement ticket around this in the future.

**Comment by tomkdgun** [comment]
Fix was not provided. Please reopen this issue.
